faker/providers/company/ja_JP/__init__.py

```diff
@@ -5,10 +5,15 @@
 
 class Provider(CompanyProvider):
     formats = (
-        '{{company_prefix}} {{last_name}}',
+        '{{company_prefix}}{{last_name}}{{category}}',
+        '{{last_name}}{{category}}{{company_prefix}}',
     )
 
     company_prefixes = ('株式会社', '有限会社', '合同会社')
+    categories = ('水産', '農林', '鉱業', '建設', '食品', '印刷', '電気', 'ガス', '情報', '通信', '運輸', '銀行', '保険')
 
     def company_prefix(self):
         return self.random_element(self.company_prefixes)
+
+    def category(self):
```

Comment:
Unfortunately, `category` is too generic of a word.

The problem is that in the Faker API, provider methods are not scoped by their provider, but they are all accessed from the root `Faker` instance. This means that people will call `faker.category()`, and I'm not sure we should return a list of categories that are restricted to companies.

Maybe we can just call it `company_category`?

Reply:
Thanks. I understand it.
Just renamed them `company_category` and `company_categories`